core/install.js

```diff
@@ -70,12 +72,22 @@ let exec = require('child_process').execSync,
 
 exports.requireOrInstall = function (req, name, dirName) {
     coffeescriptRequireInjector();
+    let r;
     if (resolve(name, dirName)) {
-        return req(name);
+        if (nativeReloadHacksCache.hasOwnProperty(name)) {
+            return nativeReloadHacksCache[name];
+        }
+        r = req(name);
+    }
+    else {
+        install(name);
+        r = req(name);
     }
 
-    install(name);
-    return req(name);
+    if (nativeReloadHacks.includes(name) || name.endsWith('.node')) {
+        nativeReloadHacksCache[name] = r;
+    }
+    return r;
 };
```

Comment:
Very convoluted logic. This might be better:
```js
if (!resolve(name, dirName)) {
	install(name);
}
else if (nativeReloadHacksCache.hasOwnProperty(name)) {
	return nativeReloadHacksCache[name];
}
let r = req(name);
```